### PR TITLE
[serve] document auto num replicas

### PR DESCRIPTION
**Pull request by zcin**
[serve] document auto num replicas

Document setting `num_replicas="auto"`. Change example in basic autoscaling guide to use `num_replicas=auto`.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>

### CODE REVIEW

File: doc/source/serve/autoscaling-guide.md [hunk 1]
```diff
@@ -22,29 +22,40 @@ deployments:
 
 ## Autoscaling Basic Configuration
 
-Instead of setting a fixed number of replicas for a deployment and manually updating it, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler reacts to traffic spikes by monitoring queue sizes and making scaling decisions to add or remove replicas. Configure it by setting the [autoscaling_config](../serve/api/doc/ray.serve.config.AutoscalingConfig.rst) field in deployment options. The following is a quick guide on how to configure autoscaling.
+Instead of setting a fixed number of replicas for a deployment and manually updating it, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler reacts to traffic spikes by monitoring queue sizes and making scaling decisions to add or remove replicas. Turn on autoscaling for a deployment by setting `num_replicas=auto`. You can further configure it by tuning the [autoscaling_config](../serve/api/doc/ray.serve.config.AutoscalingConfig.rst) in deployment options.
```
Comment (GeneDer) [inline doc/source/serve/autoscaling-guide.md:25]: Non-blocker: Maybe we can do `num_replicas="auto"` instead so the user know this is a literal "auto" and not an enum that they need to be imported from somewhere (when using it on the deployment decorator). Also would be nice to have an example of the actual deployment decorator using it.

Same goes to other places doing `num_replicas=auto`.

File: doc/source/serve/autoscaling-guide.md [hunk 2]
```diff
@@ -84,13 +90,13 @@ The results of the load test are as follows:
 |  |  |  |
 | -------- | --- | ------- |
 | Replicas | <img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_replicas.png" alt="replicas" width="600"/> |
-| QPS | <img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_rps.svg" alt="qps"/> |
-| P50 Latency | <img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_latency.svg" alt="latency"/> |
+| QPS | <img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_rps.png" alt="qps"/> |
```
Comment (GeneDer) [inline doc/source/serve/autoscaling-guide.md:93]: Non-blocker: Would be nice if we can save those as svg 🙃

**Review by GeneDer — APPROVED** [review]
LGTM!

**Review by akshay-anyscale — APPROVED** [review]

File: doc/source/serve/autoscaling-guide.md (hunk 1 above)
Comment (zcin) [reply doc/source/serve/autoscaling-guide.md]: Done! The example of an actual deployment decorator is there already, it's part of the code block from a python file :D
Screenshot 2024-03-15 at 12 12 49 PM

**Review by edoakes — APPROVED** [review]

**Pull request merged**